Artesian/Artesian.SDK/Service/Clients/Client.cs

```diff
@@ -147,21 +148,59 @@ public Client(IArtesianServiceConfig config, string Url, ArtesianPolicyConfig po
                         {
                             var responseText = await res.Content.ReadAsStringAsync();
                             var exceptionMessage = $"Failed handling REST call to WebInterface {method} {_url + resource}. Returned status: {res.StatusCode}. Content: \n";
+                            var responseDeserialized = await res.Content.ReadAsAsync<object>(_formatters, ctk);
 
                             if (res.StatusCode == HttpStatusCode.BadRequest)
                             {
-                                var responseDeserialized = await res.Content.ReadAsAsync<object>(_formatters, ctk);
+                                if (res.Content.Headers.ContentType.MediaType == "application/problem+json")
+                                {
+                                    ArtesianSdkProblemDetail problemDetail = JsonConvert.DeserializeObject<ArtesianSdkProblemDetail>(responseDeserialized.ToString());
+
+                                    if (problemDetail.Detail != null)
+                                    {
+                                        responseText = problemDetail.Detail;
+                                    }
+                                    else if (problemDetail.Extensions["Errors"].Count() > 0)
```

Comment:
Do not peek into Extensions, is too dangerous. You have no idea how is make inside.
Also this breaks if Extension["Errors"] is not a collection, since you call "Count()" over it.
What happens if Extensions["Errors"] is null?

Also the line below Extensions["Errors"][i]["Value"][0]["ErrorMessage"] is only safe on `i`, you have not guarantee "Value" exsits, is an array and contains an "ErrorMessage".

Please just drop this code and just write `responseText = problemDetails.Detail ?? problemDetails.Title ?? problemDetails.Type;`

Reply:
It had seemed if it was of the ValidationProblemDetails model an empty "error" would return an empty array. But still unknowable if this case is unique. Agree its safer to remove error message check as before.